scripts/esbuilder/esbuilder.js

```diff
@@ -51,4 +51,10 @@ build({
   define: {
     PLAYER_VERSION: `"${process.env.npm_package_version}"`,
   },
-}).catch(() => process.exit(1));
+};
+
+if (options.format === 'esm' || options.format === 'cjs') {
+  options.external = ['@mux-elements/*', '@github/template-parts', 'media-chrome'];
```

Comment:
switch to non-blocking but I'm not fully convinced. 

we're assuming devs will use these features of the CDN's. it might not work with all the NPM cdn's.

from unpkg.com:

> Expands all [“bare” import specifiers](https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier) in JavaScript modules to unpkg URLs. This feature is very experimental

Reply:
Yeah, it's definitely a bit complicated and there isn't one answer for the esm builds. The decision of what we do basically depends on whom we're trying to target with these build files.
Personally, I think they should target bundler users (or users with some form of build-step) rather than native esm because native support just isn't quite there right now, and I do believe the majority of users are still using bundlers.

But there's definitely a case to be made to be able to support native esm. However, to do that we'd want to consider bundling all dependencies, which means that then shared dependencies can't be shared between us and any code that depends on the same module. It's trade-offs all the way down.

Reply:
Also, due to some of the other issues I've brought up, that will actually affect this too. Basically, it involves the fact that things like media-chrome and template-parts are esm-only but hls.js and mux-embed are cjs-only. Making both builds and tests work properly is interesting.
I'll have a better write-up on that at some point so that we can discuss it in further detail.

Reply:
Yeah I'd recommend treating this as a followup. I don't think it would be bad for some flavor of "why not both?" (insert adorable child here), but I believe these changes are a big improvement wrt the status quo. We had discussed e.g. providing both a "TS -> JS compiled but unbundled" & a fully bundled ESM version previously as an option, but I don't think that should be a blocker here.